File: src/main/java/factorio/debugger/DAP/messages/response/DAPDisassembleResponse.java
package factorio.debugger.DAP.messages.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;
import factorio.debugger.DAP.messages.DAPResponse;
import factorio.debugger.DAP.messages.types.DAPDisassembledInstruction;

@JsonTypeName("disassemble")
public class DAPDisassembleResponse extends DAPResponse {
    @JsonProperty("body")
    public DisassembleResponseBody body;
    public static class DisassembleResponseBody {
        /**
         * The list of disassembled instructions.
         */
        @JsonProperty("instructions")
        public DAPDisassembledInstruction[] instructions;
    }
}

File: src/main/kotlin/factorio/debugger/DAP/messages/responses/DAPDisassembleResponse.kt
package factorio.debugger.DAP.messages.responses

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonTypeName
import factorio.debugger.DAP.messages.DAPAdditionalProperties
import factorio.debugger.DAP.messages.types.DAPDisassembledInstruction

@JsonTypeName("disassemble")
class DAPDisassembleResponse : DAPResponse() {
    @JsonProperty("body")
    var body: DisassembleResponseBody? = null

    class DisassembleResponseBody : DAPAdditionalProperties() {
        /**
         * The list of disassembled instructions.
         */
        @JsonProperty("instructions")
        lateinit var instructions: Array<DAPDisassembledInstruction>
    }
}
